### ISSUE TITLE
validate the statistical information periodically and when modifying it

### ISSUE THREAD

**Issue by qw4990** [pingcap/tidb#20215]
## Feature Request

**Is your feature request related to a problem? Please describe:**

Now we have met several cases caused by damaged statistical information.
Many reasons may cause this problem, bugs in TiDB and other systems, invalid operations processed by users...

**Describe the feature you'd like:**

It's better to let TiDB validate the statistical information periodically and when modifying it.
When damaged statistical information is detected, log some error information and label it as `pseudo`.
This may help us investigate statistical information problems and avoid some sever problems.

**Comment by qw4990** [comment]
What're your opinions? @zz-jason @eurekaka @winoros @SunRunAway

**Comment by zz-jason** [comment]
once the stats is damaged, how to detect the endless loop and quit it?

**Comment by qw4990** [comment]
> once the stats is damaged, how to detect the endless loop and quit it?

We can check if all buckets' left bounds are larger than the previous bucket's right bound to avoid endless loop. @zz-jason

**Comment by zz-jason** [comment]
Yes, but it requires to check every loop on the histogram buckets, am I correct? Could you describe more implementation detail on this issue, for example, does it need a dedicated goroutine to validate the correctness of the histogram, does it also need to validate the histogram in every `for` loop on the histogram?
